### ISSUE TITLE
Update Input component to comply with Material design and update all input fields to use component

### ISSUE THREAD

**Issue by m-deacon** [rehive/wallet-react-native#80]

**Comment by m-deacon** [comment]
Hi @Cupofassam - could you please review the new look.

Once this is ok'd and #76 as well, I'll pull 76 into this branch and roll this change to all other components (otherwise there will be conflicts as file names have been changed.

**Comment by Cupofassam** [comment]
Some feedback:

1. Material design advises helper text below fields.I think it would be useful to add helper text below the password fields on the registration page to let users know that their passwords have to be a minimum pf 8 characters.

![image](https://user-images.githubusercontent.com/22153546/39621573-89613e0e-4f8f-11e8-8d37-771b0ba92e5f.png)

2. On the registration page there is quite a bit gap between the flag and the +1 as well as a random grey line - could we reduce the spacing and remove the line?

![image](https://user-images.githubusercontent.com/22153546/39621619-b5853cc4-4f8f-11e8-8cc1-061514c6235c.png)

3. For errors, I think we need to adjust a few things:

- Padding between line and error should increase a bit
- The text input lines should be red instead of grey
- We should add "Error:"before the error message to improve accessibility

![image](https://user-images.githubusercontent.com/22153546/39622619-3b71ebf4-4f93-11e8-9305-0ae7043b695e.png)

4. On the login screen I feel like the button is a bit too close to the password field - can we drop it down so that the spacing between the password field and the button is the same as between the button and the Forgot password link?

![image](https://user-images.githubusercontent.com/22153546/39622808-c9213bc6-4f93-11e8-9750-aa81601ed0dc.png)

**Comment by m-deacon** [comment]
Hi @Cupofassam - I've made the above changes.

Things to note:
1. I've only added helper text to the password fields, not sure if it's needed on the others?
2. The reason the code was so far away was to cater for long 4 digit dialling codes. I've updated this to be variable depending on the selected code.
3. Personally I'm not the biggest fan of the changing underline to red when there is an error, but I've added it. Perhaps it will be better when we split up the register screen as I'm busy doing.
